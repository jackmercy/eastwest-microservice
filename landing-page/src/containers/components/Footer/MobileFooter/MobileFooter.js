import React, { memo, useMemo, useState } from 'react';
import Link from 'next/link';
import { useIntl } from 'react-intl';
import classNames from 'classnames';

import {
  Register,
  CopyRight,
  SummaryInfo,
  SectionTitle,
  SwitchLanguages,
} from '../components';

import useStyles from './mobileFooter.styles';

const MobileFooter = () => {
  const intl = useIntl();
  const classes = useStyles();
  const [activatedIndex, setActiveIndex] = useState(-1);
  const configs = useMemo(
    () => [
      {
        key: 'product',
        title: intl.formatMessage({ id: 'FOOTER.PRODUCTS.TITLE' }),
        dataTest: 'footer-product',
        menu: [
          {
            key: 'talent-labs',
            href: '/talent-labs',
            label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.TALENT_LABS' }),
          },
          // MVP hidden
          // {
          //   key: 'dashboard',
          //   href: '/dashboard',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.DASHBOARD' }),
          // },
          // {
          //   key: 'art',
          //   href: '/art',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.ART' }),
          // },
          // {
          //   key: 'admin',
          //   href: '/admin',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.ADMIN' }),
          // },
          // {
          //   key: 'rmt',
          //   href: '/rmt',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.RMT' }),
          // },
          {
            key: 'data-labs',
            href: '/data-labs',
            label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.DATA_LABS' }),
          },
        ],
      },
      {
        key: 'company',
        title: intl.formatMessage({ id: 'FOOTER.COMPANY.TITLE' }),
        dataTest: 'footer-page',
        menu: [
          {
            key: 'about-us',
            href: '/company',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.ABOUT_US' }),
          },
          {
            key: 'careers',
            href: '/careers',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.CAREERS' }),
          },
          {
            key: 'contact',
            href: '/contact',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.CONTACT_US' }),
          },
        ],
      },
      {
        key: 'learn',
        title: intl.formatMessage({ id: 'FOOTER.LEARN.TITLE' }),
        dataTest: 'footer-other',
        menu: [
          {
            key: 'news',
            href: '/news-and-media',
            label: intl.formatMessage({ id: 'FOOTER.LEARN.NEWS' }),
          },
          // MVP hidden
          // {
          //   key: 'support',
          //   href: '/support',
          //   label: intl.formatMessage({ id: 'FOOTER.LEARN.SUPPORT' }),
          // },
          // {
          //   key: 'newsletter',
          //   href: '/newsletter',
          //   label: intl.formatMessage({ id: 'FOOTER.LEARN.NEWSLETTER' }),
          // },
        ],
      },
    ],
    [intl],
  );

  const handleItemClick = (index) => {
    setActiveIndex(index !== activatedIndex ? index : -1);
  };

  return (
    <footer className={classes.root}>
      <SummaryInfo />
      {configs.map(({ key: parentKey, title, dataTest, menu }, index) => (
        <div key={parentKey} className={classes.dropdown}>
          <div
            role="presentation"
            className={classes.toggleButton}
            onClick={() => handleItemClick(index)}
          >
            <SectionTitle title={title} />
            <i
              className={
                index === activatedIndex ? 'icon-caret-up' : 'icon-caret-down'
              }
            />
          </div>
          <ul
            className={classNames(classes.menu, {
              active: index === activatedIndex,
            })}
          >
            {menu.map(({ key, label, href, ...restProps }) => (
              <li key={`${parentKey}-${key}`} className={classes.menuItem}>
                <Link key={`${parentKey}-${key}`} href={href} passHref>
                  <a
                    href="/#"
                    className={classes.menuLink}
                    data-test={dataTest}
                    {...restProps}
                  >
                    {label}
                  </a>
                </Link>
              </li>
            ))}
          </ul>
        </div>
      ))}
      <Register />
      <SwitchLanguages />
      <CopyRight />
    </footer>
  );
};

export default memo(MobileFooter);
