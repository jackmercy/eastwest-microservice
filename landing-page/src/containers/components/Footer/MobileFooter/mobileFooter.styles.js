import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';

export default createUseStyles({
  root: {
    backgroundColor: COLORS.bgBlack01,
    width: '100%',
    padding: [rem(48), rem(32), 0],
    boxSizing: 'border-box',
    overflow: 'hidden',
  },
  dropdown: {
    margin: [0, rem(-32)],
    padding: [0, rem(32)],
    backgroundColor: COLORS.bgBlack,
  },
  toggleButton: {
    height: rem(48),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    '& > [class^=icon-]': { color: COLORS.textWhite, fontSize: rem(12) },
  },
  menu: {
    margin: [0, rem(-16)],
    padding: 0,
    maxHeight: 0,
    listStyleType: 'none',
    overflow: 'hidden',
    transition: 'max-height .5s ease',
    '&.active': { maxHeight: rem(500) },
  },
  menuItem: {
    padding: [0, rem(16)],
    display: 'flex',
    alignItems: 'center',
    height: rem(48),
    '&:not(:last-child)': {
      borderBottom: `1px solid ${hexToRgbA(COLORS.borderWhite, 0.5)}`,
    },
  },
  menuLink: {
    fontWeight: 'bold',
    fontSize: rem(14),
    lineHeight: rem(24),
    color: hexToRgbA(COLORS.textWhite, 0.8),
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    height: '100%',
    textDecoration: 'none',
  },
});
