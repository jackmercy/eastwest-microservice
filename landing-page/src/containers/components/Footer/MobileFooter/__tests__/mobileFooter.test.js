import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  Register,
  CopyRight,
  SummaryInfo,
  SwitchLanguages,
} from '../../components';
import MobileFooter from '../MobileFooter';

describe('Testing Footer > MobileFooter', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<MobileFooter />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<MobileFooter />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have SummaryInfo', () => {
    expect(wrapper.find(SummaryInfo)).toHaveLength(1);
  });

  it('should have some product urls', () => {
    expect(wrapper.find('[data-test="footer-product"]')).toHaveLength(2);
  });

  it('should have some page urls', () => {
    expect(wrapper.find('[data-test="footer-page"]')).toHaveLength(3);
  });

  // MVP hidden
  // it('should have some other urls', () => {
  //   expect(wrapper.find('[data-test="footer-other"]')).toHaveLength(3);
  // });

  it('should have a SwitchLanguages', () => {
    expect(wrapper.find(SwitchLanguages)).toHaveLength(1);
  });

  it('should have a Register', () => {
    expect(wrapper.find(Register)).toHaveLength(1);
  });

  it('should have a CopyRight', () => {
    expect(wrapper.find(CopyRight)).toHaveLength(1);
  });
});
