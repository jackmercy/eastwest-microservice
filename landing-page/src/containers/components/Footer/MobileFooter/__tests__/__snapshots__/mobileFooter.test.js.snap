// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer > MobileFooter should render match snapshot 1`] = `
<footer
  className="root-0-2-1"
>
  <Memo(SummaryInfo) />
  <div
    className="dropdown-0-2-2"
    key="product"
  >
    <div
      className="toggleButton-0-2-3"
      onClick={[Function]}
      role="presentation"
    >
      <Memo(SectionTitle)
        title="Products"
      />
      <i
        className="icon-caret-down"
      />
    </div>
    <ul
      className="menu-0-2-4"
    >
      <li
        className="menuItem-0-2-5"
        key="product-talent-labs"
      >
        <Link
          href="/talent-labs"
          key="product-talent-labs"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-product"
            href="/#"
          >
            Talent labs
          </a>
        </Link>
      </li>
      <li
        className="menuItem-0-2-5"
        key="product-data-labs"
      >
        <Link
          href="/data-labs"
          key="product-data-labs"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-product"
            href="/#"
          >
            Data labs
          </a>
        </Link>
      </li>
    </ul>
  </div>
  <div
    className="dropdown-0-2-2"
    key="company"
  >
    <div
      className="toggleButton-0-2-3"
      onClick={[Function]}
      role="presentation"
    >
      <Memo(SectionTitle)
        title="Company"
      />
      <i
        className="icon-caret-down"
      />
    </div>
    <ul
      className="menu-0-2-4"
    >
      <li
        className="menuItem-0-2-5"
        key="company-about-us"
      >
        <Link
          href="/company"
          key="company-about-us"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-page"
            href="/#"
          >
            About us
          </a>
        </Link>
      </li>
      <li
        className="menuItem-0-2-5"
        key="company-careers"
      >
        <Link
          href="/careers"
          key="company-careers"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-page"
            href="/#"
          >
            Careers
          </a>
        </Link>
      </li>
      <li
        className="menuItem-0-2-5"
        key="company-contact"
      >
        <Link
          href="/contact"
          key="company-contact"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-page"
            href="/#"
          >
            Contact us
          </a>
        </Link>
      </li>
    </ul>
  </div>
  <div
    className="dropdown-0-2-2"
    key="learn"
  >
    <div
      className="toggleButton-0-2-3"
      onClick={[Function]}
      role="presentation"
    >
      <Memo(SectionTitle)
        title="Learn"
      />
      <i
        className="icon-caret-down"
      />
    </div>
    <ul
      className="menu-0-2-4"
    >
      <li
        className="menuItem-0-2-5"
        key="learn-news"
      >
        <Link
          href="/news-and-media"
          key="learn-news"
          passHref={true}
        >
          <a
            className="menuLink-0-2-6"
            data-test="footer-other"
            href="/#"
          >
            News & Media
          </a>
        </Link>
      </li>
    </ul>
  </div>
  <Memo(Register) />
  <Memo(SwitchLanguages) />
  <Memo(CopyRight) />
</footer>
`;
